src/Suggestor/EditorSuggestorPopup.ts

```diff
@@ -73,6 +89,14 @@ export class EditorSuggestor extends EditorSuggest<SuggestInfoWithContext> {
             (editor as any)?.cm?.contentDOM?.dispatchEvent(eventClone);
             return;
         }
+
+        if (value.taskItDependsOn != null) {
+            const newTask = ensureTaskHasId(value.taskItDependsOn, ['TODO']);
```

Comment:
A comment explaining the TODO would be really useful.

Reply:
I think this behaviour is going to be a problem. That second argument is used to ensure that no duplicate ids are created - by not passing in the actual task list, there is a (small) risk that a duplicate id may be created.

Reply:
Indeed, i know i had added a Second TODO somewhere :)

I admit, i forgot it. Sorry. Fixed it.
Good thing 4 Eyes see more, than Two.